dev/dev-env-setup.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+
+set +exv
+
+showHelp() {
+cat << EOF
+Usage: ./install-dev-env.sh [-d] [directory to install virtual environment] [-v] [-q]
+Development environment setup script for Python.
+This script will:
+
+  - Install pyenv if not installed
+  - Retrieve the appropriate Python version (minimum required) for compatibility support
+  - Check if the virtual environment already exists
+    - If it does, prompt for replacement
+      - If replacing, delete old virtual environment.
+  - Create a virtual environment using the minimum required Python version based on previous step logic
+  - Activate the environment
+  - Install required dependencies for the dev envrionment
+
+  Example usage:
+
+  From root of MLflow repository on local with a destination virtualenv path of <root>/.venvs/mlflow-dev:
+
+  dev/dev-env-setup.sh -d ~/.venvs/mlflow-dev
+
+  Note: it is recommended to preface virtualenv locations with a directory name prefaced by '.' (i.e., ".venvs").
+
+-h, -help,        --help        Display help
+
+-d, -directory    --directory   The path to install the virtual environment into
+
+-v, -verbose      --verbose     Whether to fill stdout with every command or not
+
+-q, -quiet        --quiet       Whether to have pip install in quiet mode (Default: false)
+
+EOF
+}
+
+export verbose=0
+while getopts "d:vqh" opt
+do
+  case "$opt" in
+    d) directory="$OPTARG" ;;
+    v) verbose=1 ;;
+    q) quiet=1 ;;
+    h) showHelp; exit ;;
+    *) showHelp; exit ;;
+  esac
+done
+
+if [[ $verbose = 1 ]]; then
+  set -exv
+fi
+
+# Check if pyenv is installed and offer to install it if not present
+pyenv_exist=$(command -v pyenv)
+
+if [ -z "$pyenv_exist" ]; then
+  read -p "pyenv is required to be installed to manage python versions. Would you like to install it? $(tput bold)(y/n)$(tput sgr0): " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    echo "Updating brew and installing pyenv..."
+    echo "Note: this will probably take a considerable amount of time."
+    brew update
+    brew install pyenv
```

Comment:
Does this script only support macOS?

Reply:
great point. I'll add the pyenv setup for a few OS's.

Reply:
Can we just provide the pyenv installation instructions and let contributors install pyenv by themselves?

Reply:
I'm concerned about maintenance costs. If we support a few OS, I think we need to test the script on the supported operating systems.